server/py/framework/db/sqldb/db.py

```diff
@@ -5929,14 +5930,67 @@ def store_alert_activation(
         self._upsert(session, [alert_activation_record])
 
     def list_alert_activations(
-        self, session: Session, project: typing.Optional[str] = None
+        self,
+        session: Session,
+        projects_with_creation_time: list[tuple[str, datetime]],
+        name: typing.Optional[str] = None,
+        since: typing.Optional[str] = None,
+        until: typing.Optional[str] = None,
+        entity: typing.Optional[str] = None,
+        severity: Optional[
+            list[Union[mlrun.common.schemas.alert.AlertSeverity, str]]
+        ] = None,
+        entity_kind: Optional[
+            Union[mlrun.common.schemas.alert.EventEntityKind, str]
+        ] = None,
+        event_kind: Optional[Union[mlrun.common.schemas.alert.EventKind, str]] = None,
+        page: typing.Optional[int] = None,
+        page_size: typing.Optional[int] = None,
     ) -> list[mlrun.common.schemas.AlertActivation]:
-        # TODO: add filters
         query = self._query(session, AlertActivation)
 
-        if project and project != "*":
-            query = query.filter(AlertActivation.project == project)
+        # Filter alert activations for the project created after the project creation date,
+        # excluding activations linked to any previous instances of the project.
+        # TODO: reconsider this approach when we move alerts out of main MLRun db
+        conditions = [
```

Comment:
Just for explicitness
```suggestion
        project_filter_conditions = [
```